scipy/special/add_newdocs.py

```diff
@@ -8890,7 +8890,7 @@ def add_newdoc(name, doc):
     It is defined to be
 
     .. math::
-      \int_0^z \frac{\log(t)}{1 - t}dt
+      \int_1^z \frac{\log(t)}{1 - t}dt
```

Comment:
This correction makes sense to me & matches the wikipedia formulation. A few pieces of evidence below, beyond those in the original issue:

1) `scipy/special/cephes/spence.c ` documentation says so:

![image](https://user-images.githubusercontent.com/7903078/118404549-a05bfa80-b630-11eb-8533-a85ddd39f4ce.png)

2) `scipy/special/cephes/spence.c ` implementation seems to enforce the 1-based integration with an explicit 0 return in that case:

![image](https://user-images.githubusercontent.com/7903078/118404611-d7caa700-b630-11eb-9b3f-bf32b433c712.png)